src/scale_impls/encode.rs

```diff
@@ -421,37 +423,19 @@ mod test {
             Unnamed(u64, Vec<bool>),
         }
 
-        let named_value = Value::named_variant(
-            "Named",
-            [
-                // Deliverately a different order; order shouldn't matter:
-                ("foo", Value::bool(true)),
-                ("hello", Value::string("world")),
-            ],
-        );
+        let named_value = value!(Named { foo: true, hello: "world" });
+
         assert_can_encode_to_type(named_value, Foo::Named { hello: "world".into(), foo: true });
 
-        let unnamed_value = Value::unnamed_variant(
-            "Unnamed",
-            [
-                Value::u128(123),
-                Value::unnamed_composite(vec![
-                    Value::bool(true),
-                    Value::bool(false),
-                    Value::bool(true),
-                ]),
-            ],
-        );
+        let unnamed_value = value!(Unnamed(123u64, (true, false, true)));
```

Comment:
Out of interest, what happens if the type of the number isn't given (ie no `u64`)? (I might have to try this!)

All numbers become u128's anyway so in theory these types don't help with anything

Reply:
It does not matter much, if I use either u8, u16, ... or u64 here, but it has to be an unsigned integer. Because only unsigned integers will result in `Primitive::U128(u128)`. Otherwise, e.g. just saying `123` will make it a i32, which gets converted into a `Primitive::I128(i128)` which makes the test fail.